### PR TITLE
add Pure performance

### PR DESCRIPTION
**Pull request by zhilong-xu**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: delfin/drivers/pure/flasharray/pure_flasharray.py [hunk 1]
```diff
@@ -363,6 +364,134 @@ def reset_connection(self, context, **kwargs):
...
+                start_time, end_time,
+                self.rest_handler.REST_VOLUME_HISTORICAL_URL,
+                constants.ResourceType.VOLUME)
+            metrics.extend(storage_metrics)
+        return metrics
+
+    def get_metrics(self, storage_id, resource_metrics, start_time, end_time,
+                    url, resource_type, resource_id=None, resource_name=None):
+        metrics = []
+        localtime = time.mktime(time.localtime())
```
Comment (ghxiaobo) [inline delfin/drivers/pure/flasharray/pure_flasharray.py:392]: Add an underscore, such as local_time

File: delfin/drivers/pure/flasharray/pure_flasharray.py [hunk 2]
```diff
@@ -363,6 +364,134 @@ def reset_connection(self, context, **kwargs):
...
+                    'resource_id': resource_id,
+                    'resource_name': resource_name,
+                    'type': 'RAW',
+                    'unit': resource_metrics[resource_key]['unit']
+                }
+                resource_value = {timestamp: metrics_data.get(resource_key)}
+                metrics_res = constants.metric_struct(name=resource_key,
+                                                      labels=labels,
+                                                      values=resource_value)
+                metrics.append(metrics_res)
```
Comment (ghxiaobo) [inline delfin/drivers/pure/flasharray/pure_flasharray.py:441]: This method code is too long

File: delfin/drivers/pure/flasharray/pure_flasharray.py [hunk 3]
```diff
@@ -363,6 +364,134 @@ def reset_connection(self, context, **kwargs):
...
+            historical = consts.ONE_DAY
+        if start_time < localtime - consts.ONE_DAY_DIFFERENCE:
+            historical = consts.SEVEN_DAY
+        if start_time < localtime - consts.SEVEN_DAY_DIFFERENCE:
+            historical = consts.THIRTY_DAY
+        if start_time < localtime - consts.THIRTY_DAY_DIFFERENCE:
+            historical = consts.NINETY_DAY
+        if start_time < localtime - consts.NINETY_DAY_DIFFERENCE:
+            historical = consts.ONE_YEAR
+        return historical
```
Comment (ghxiaobo) [inline delfin/drivers/pure/flasharray/pure_flasharray.py:469]: use elif

File: delfin/drivers/pure/flasharray/pure_flasharray.py (hunk 1 above)
Comment (zhilong-xu) [reply delfin/drivers/pure/flasharray/pure_flasharray.py:392]: The name of a method that cannot be changed

Comment (zhilong-xu) [reply delfin/drivers/pure/flasharray/pure_flasharray.py:392]: Has been modified

Comment (zhilong-xu) [reply delfin/drivers/pure/flasharray/pure_flasharray.py:392]: Has been modified

local_time = time.mktime(time.localtime())

File: delfin/drivers/pure/flasharray/pure_flasharray.py (hunk 2 above)
Comment (zhilong-xu) [reply delfin/drivers/pure/flasharray/pure_flasharray.py:441]: Has been modified

metrics_res = constants.metric_struct(
                    name=resource_key, labels=labels, values=resource_value)

File: delfin/drivers/pure/flasharray/pure_flasharray.py (hunk 3 above)
Comment (zhilong-xu) [reply delfin/drivers/pure/flasharray/pure_flasharray.py:469]: Has been modified

if start_time < localtime - consts.NINETY_DAY_DIFFERENCE:
            historical = consts.ONE_YEAR
        elif start_time < localtime - consts.THIRTY_DAY_DIFFERENCE:
            historical = consts.NINETY_DAY
        elif start_time < localtime - consts.SEVEN_DAY_DIFFERENCE:
            historical = consts.THIRTY_DAY
        elif start_time < localtime - consts.ONE_DAY_DIFFERENCE:
            historical = consts.SEVEN_DAY
        elif start_time < localtime - consts.THREE_HOUR_DIFFERENCE:
            historical = consts.ONE_DAY
        elif start_time < localtime - consts.ONE_HOUR_DIFFERENCE:
            historical = consts.THREE_HOUR
        else:
            historical = consts.ONE_HOUR

**Pull request merged**